src/test/regress/spec/isolation_create_distributed_concurrently_after_drop_column.spec

```diff
@@ -0,0 +1,95 @@
+#include "isolation_mx_common.include.spec"
+
+// Test scenario for nonblocking split and concurrent INSERT/UPDATE/DELETE
+//  session s1 - Executes create_distributed_table_concurrently after dropping a column
+//  session s2 - Does concurrent inserts
+//  session s3 - Holds advisory locks
+
+setup
+{
+	SET citus.shard_replication_factor TO 1;
+	CREATE TABLE observations (
+  	tenant_id text not null,
+  	dummy int,
+  	measurement_id bigserial not null,
+  	payload jsonb not null,
+  	observation_time timestamptz not null default now(),
+  	PRIMARY KEY (tenant_id, measurement_id)
+	);
+}
+
+teardown
+{
+    DROP TABLE observations;
+}
+
+session "s1"
+
+step "s1-alter-table"
+{
+	ALTER TABLE observations DROP COLUMN dummy;
+}
+
+step "s1-set-factor-1"
+{
+	SET citus.shard_replication_factor TO 1;
+	SELECT citus_set_coordinator_host('localhost');
+}
+
+step "s1-create-distributed-table-concurrently"
+{
+	SELECT create_distributed_table_concurrently('observations','tenant_id');
+}
+
+session "s2"
+
+step "s2-begin"
+{
+    BEGIN;
+}
+
+step "s2-insert"
+{
+	INSERT INTO observations(tenant_id, payload) SELECT 'tenant_id', jsonb_build_object('name', 29.3);
+	INSERT INTO observations(tenant_id, payload) SELECT 'tenant_id', jsonb_build_object('name', 29.3);
+	INSERT INTO observations(tenant_id, payload) SELECT 'tenant_id', jsonb_build_object('name', 29.3);
```

Comment:
would be good to also add some update and delete tests with replica identity full

Reply:
@marcocitus : Yep. I have added locally the two combinations: Update + Delete on replicat identities with PK and FULL. 

Pushing in a bit.

Reply:
Pushed the latest change set with coverage and fix of replica identities!